api/test/round-tracker.test.js

```diff
@@ -626,6 +626,131 @@ describe('Round Tracker', () => {
             clients: ['f0050']
           }
         ])
+
+        // Clean up
+        await pgClient.query('DELETE FROM retrieval_tasks WHERE round_id = $1', [roundId])
+        await pgClient.query('DELETE FROM spark_rounds WHERE id = $1', [roundId])
+        await pgClient.query("DELETE FROM eligible_deals WHERE miner_id LIKE 'f001%' AND client_id = 'f0050' AND piece_cid LIKE 'baga%' AND piece_size = 1")
+      })
+
+      it('should handle single allocator with multiple clients correctly', async () => {
+        // Insert test data with one allocator having multiple clients
+        await pgClient.query(`
+          INSERT INTO eligible_deals
+          (miner_id, client_id, piece_cid, piece_size, payload_cid, expires_at)
+          VALUES
+          ('f0020', 'clientA1', 'baga1', 1, 'bafkTest', NOW() + INTERVAL '1 year'),
+          ('f0021', 'clientA2', 'baga2', 1, 'bafkTest', NOW() + INTERVAL '1 year'),
+          ('f0022', 'clientA3', 'baga3', 1, 'bafkTest', NOW() + INTERVAL '1 year')
+        `)
+
+        await pgClient.query(`
+          INSERT INTO allocator_clients (client_id, allocator_id)
+          VALUES 
+            ('clientA1', 'allocator'),
+            ('clientA2', 'allocator'),
+            ('clientA3', 'allocator')
+        `)
+
+        const roundNumber = 10024n
+        await pgClient.query(`
+          INSERT INTO spark_rounds
+          (id, created_at, meridian_address, meridian_round, start_epoch, max_tasks_per_node)
+          VALUES
+          ($1, NOW(), '0x1a', 1, 1, 15)
+        `, [
+          roundNumber
+        ])
+
+        await defineTasksForRound(pgClient, roundNumber, 10)
```

Comment:
Since you have only three eligible deals to choose retrieval tasks from, calling `defineTasksForRound` with task count 10 may introduce a test failure in the future, if defineTasksForRound starts to check that it produced the right number of tasks.

Let's use task count 3 instead - that's what you want in this test, isn't it?

```suggestion
        await defineTasksForRound(pgClient, roundNumber, 3)
```

The same comment applies to the other test below too.